src/__experimental__/components/date-range/date-range.component.js

```diff
@@ -0,0 +1,193 @@
+
+    /**
+     * Custom message for startDate field
+     */
+    startMessage: PropTypes.string,
+
+    /**
+     * Custom message for endDate field
+     */
+    endMessage: PropTypes.string,
+
+    /**
+     * Display labels inline
+     */
+    labelsInline: PropTypes.bool,
+
+    /**
+     * Props for the child start Date component
+     */
+    startDateProps: PropTypes.shape({ ...Date.propTypes, value: PropTypes.string }),
+
+    /**
+     * Props for the child end Date component
+     */
+    endDateProps: PropTypes.shape({ ...Date.propTypes, value: PropTypes.string })
+  };
+
+  /**
+   * onChange function -triggers validations on both fields and updates opposing field when one changed.
+   */
+  _onChange = (changedDate, ev) => {
+    const newValue = ev.target.value;
+
+    if (changedDate === 'startDate') {
+      this.props.onChange([newValue, this.endDate]);
+    }
+
+    if (changedDate === 'endDate') {
+      this.props.onChange([this.startDate, newValue]);
+    }
+
+    // Triggers validations on both fields
+    if (DateHelper.isValidDate(newValue)) {
+      this._startDate.handleBlur();
+      this._endDate.handleBlur();
+    }
+  }
+
+  /**
+   * The startDate value
+   */
+  get startDate() {
+    if (this.props.startDateProps && this.props.startDateProps.value) {
+      return this.props.startDateProps.value;
+    }
+    return this.props.value[0];
+  }
+
+  /**
+   * The endDate value
+   */
+  get endDate() {
+    if (this.props.endDateProps && this.props.endDateProps.value) {
+      return this.props.endDateProps.value;
+    }
+    return this.props.value[1];
+  }
+
+  /**
+   * The error message for the start message.
+   */
+  get startMessage() {
+    return this.props.startMessage
+      || I18n.t('errors.messages.date_range', { defaultValue: 'Start date must not be later than the end date' });
+  }
+
+  /**
+   * The error message for the end message.
+   */
+  get endMessage() {
+    return this.props.endMessage
+     || I18n.t('errors.messages.date_range', { defaultValue: 'End date cannot be earlier than the start date' });
+  }
+
+
+  /**
+   * Handle focus on start date field
+   */
+  focusStart = () => {
+    this._endDate.closeDatePicker();
+  }
+
+  /**
+   * Handle focus on end date field
+   */
+  focusEnd = () => {
+    this._startDate.closeDatePicker();
+  }
+
+  /**
+   * The startDate props
+   */
+  startDateProps() {
+    return this.dateProps('start', [
+      new DateRangeValidator({
+        endDate: this.endDate,
+        messageText: this.startMessage
+      })
+    ]);
+  }
+
+  /**
+   * The endDate props
+   */
+  endDateProps() {
+    return this.dateProps('end', [
+      new DateRangeValidator({
+        startDate: this.startDate,
+        messageText: this.endMessage
+      })
+    ]);
+  }
+
+  /**
+   * The startDate/endDate props
+   */
+  dateProps(propsKey, defaultValidations) {
+    const dateProps = this.props[`${propsKey}DateProps`] || {};
+
+    const props = assign({}, {
```

Comment:
there is a lot more to refactor in this component, I've opted to not change the internal implementation, that should be done as a separate ticket in the future.

Reply:
I just thought that as this was such a tiny change, and it'll remove `lodash` from the file, it might be part of normal work.

If you don't want to fix it here, could you create a ticket for it for the backlog?

Reply:
https://jira.sage.com/browse/FE-1502